src/lib/worker/drawers/drawLine.ts

```diff
@@ -36,11 +39,34 @@ export function drawLine(drawContext: DrawContext, lineAttrs: LineInfo, xBounds:
     }
 
     ctx.beginPath();
-    let penDown = false;
+    let penDown: boolean = false;
+    let startX: number | null = null;
+
+    function fillPath(startX: number, endX: number) {
+        ctx.stroke();
+        ctx.lineTo(endX, canvas.height);
+        ctx.lineTo(startX, canvas.height);
+        ctx.closePath();
+        ctx.fill();
+
+        ctx.beginPath();
+    }
```

Comment:
If I understand correctly, you are filling an area under the line as many trapezoidal segments, like this:

<img src="https://github.com/user-attachments/assets/448ee49e-26e0-4266-8cbe-a09e26cb5e22" width="300"/>


 I'm afraid that this could be slow. Have you tested the  performance of this solution on large data sets?

Wouldn't it be faster if we do only two canvas paths:
1. First, fill the area using a single `beginPath()` and `fill()` covering all the area under line
2. Then draw the line above it using the same code as we use now.

This way we will do two iterations over the points list, but only two canvas paths. I'm not sure if it will be faster or not, but it certainly worth benchmarking.

?

Reply:
The algorithm works a bit differently:

1. We traverse the line until we find a gap in the values (the line is interrupted).
2. We draw the line.
3. We place a point at the end of the line at the lower boundary of the canvas, and we place a point at the beginning of the line at the lower boundary of the canvas.
4.We close the line and fill the resulting shape.

The problem was with interrupted lines (y == null). I found a solution - to divide the line into several interrupted segments and draw them separately.

The performance test on a large dataset showed approximately the same behavior as the old implementation, but this is only a visual test. I still don't understand how to create a clear performance test comparing the two solutions.

Reply:
Aah, sorry, I haven't read the code carefully enough! I thought you are filling and stroking paths on every point, but you are doing it only at every gap!

I think this is perfectly reasonable.